### PR TITLE
[Sage-515] Sage 3 - Panel Updates

### LINKED ISSUE #515: Update Panel
## Description

Update component and subcomponents to match latest specs in Sage 3.0. The following were noted during initial audit:

- Double-check border-radius
- Add subtext block
- Consider whether header changes should also be synced in Card context.

**Dependencies**:
- [Border radius](https://github.com/Kajabi/sage-lib/issues/472)

**Links**
- Figma specs:
	- [Panel](https://www.figma.com/file/9Km09NjlZHYWsMP7EGT8tI/WIP-Sage-3-%E2%80%94-Admin-Components?node-id=3561%3A9796)
	- [Headings](https://www.figma.com/file/9Km09NjlZHYWsMP7EGT8tI/WIP-Sage-3-%E2%80%94-Admin-Components?node-id=3909%3A1134)
- [Existing component(s)](https://sage-design-system.kajabi.com/pages/component/panel)

### PR DESCRIPTION
**Pull request by FuturaExtraBold**
## Description
Updates the Panel Rails and React Component styles and markup per #515. Double-checks the border-radius is correct. A new subtext block has been added to the Panel header and subheader, that accepts arbitrary markup.

## Screenshots

|  Before  |  After  |
|--------|--------|
|image|image|

### Rails

- View [Rails component](http://localhost:4000/pages/component/panel)
- Check that component UI exists and nothing is broken
- Ensure the subtext content can be added

### React
- View [Storybook](http://localhost:4100/?path=/docs/sage-panel--default)
- Check that component UI exists and nothing is broken
- Ensure the subtext content can be added

## Testing in `kajabi-products`
(MEDIUM) Alters styling and adds markup for all Panels in Kajabi Products. Please sanity-test that Panels render correctly, including this new subtext block.

## Related
Closes #515

### CODE REVIEW

File: packages/sage-assets/lib/stylesheets/components/_panel.scss [hunk 1]
```diff
@@ -53,7 +53,13 @@
...
 .sage-panel__header,
 .sage-panel__subheader {
-  @include sage-grid-panel-row();
+  display: grid;
+  justify-content: space-between;
+  align-items: center;
+  grid-template-columns: 1fr auto;
+  grid-auto-columns: auto;
+  column-gap: sage-spacing();
+  row-gap: sage-spacing(2xs);
```
Comment (philschanely) [inline packages/sage-assets/lib/stylesheets/components/_panel.scss:62]: Based on the mocks I believe we should distinguish this `row-gap` setting as `xs` on Header and `2xs` on Subheader.

File: packages/sage-assets/lib/stylesheets/components/_panel.scss [hunk 2]
```diff
@@ -104,6 +110,11 @@
   gap: sage-spacing(card);
 }
 
+.sage-panel__subtext {
+  grid-column: 1 / -1;
+  color: sage-color(charcoal, 100);
```
Comment (philschanely) [inline packages/sage-assets/lib/stylesheets/components/_panel.scss:115]: In addition to color we may want to also provide the type spec here that should apply by default. This way (hopefully) the `p` used inside will inherit the style unless a class overrides. And of course this appears to be different between Header vs. Subheader.

**Review by philschanely — APPROVED** [review]
Looking good. I have two small final suggestions but otherwise 👍

File: packages/sage-assets/lib/stylesheets/components/_panel.scss (hunk 1 above)
Comment (FuturaExtraBold) [reply packages/sage-assets/lib/stylesheets/components/_panel.scss:62]: Will update.

File: packages/sage-assets/lib/stylesheets/components/_panel.scss (hunk 2 above)
Comment (FuturaExtraBold) [reply packages/sage-assets/lib/stylesheets/components/_panel.scss:115]: Will update.

**Review by philschanely — APPROVED** [review]
Nice work @FuturaExtraBold !

**Comment by FuturaExtraBold** [comment]
I added a `sage-panel__header--custom` class when a header has more than 1 child other than `PanelSubtext` *or* the `subtext` property is not `nil`. This adjusts the styling so that the `sage-panel__title` would fill the entire width of the header.

**Comment by FuturaExtraBold** [comment]
I also audited all instances of `SagePanelHeader` and `SagePanelSubheader`. These are files that may need a second look for sanity. All other instances use the `title` or `subtitle` property only. No existing `SagePanelSubheader`s need any work.

**Update to Use New Subtext Feature**
admin/affiliates_settings/affiliate_site_pages x 2
admin/affiliates_settings/conversion_window
admin/affiliates_settings/registrations
admin/offer_details/_limit_offer.html.erb
admin/offer_details/_payment_providers.html.erb
admin/offers/edit/_post_purchase_email.html.erb
admin/offers/edit/_post_purchase_location.html.erb

**Possibly Wonky with Help Link in Title**
admin/automation_rules/panel

**Help Link Icon Used**
admin/forms/v2/_double_opt_in_panel.html.erb
admin/reports/_activity_feed.html.erb

**Using Icon**
admin/coaching_programs/session_trackings/_agenda.html.erb
admin/coaching_programs/session_trackings/_header.html.erb
admin/product_blueprints/index.html.erb
admin/sites/index.html.erb

**Using Popover**
admin/site_custom_domains_v2/new.html.erb x 3
admin/podcasts/episodes/_publishing.html.erb

**Using Dropdown**
admin/navbars/_navbar.html.erb

**Empty Headers, Can Simplify**
admin/offers/v3/_included_products.html.erb x 2

**Using Modal Close Icon**
admin/marketing_email_templates/_modal_create_email_from_template.html.erb
admin/marketing_email_templates/_modal_template_details.html.erb
admin/website_pages/_add_website_page_modal.html.erb

File: packages/sage-react/lib/Panel/PanelHeader.jsx [hunk 3]
```diff
@@ -1,32 +1,59 @@
...
-    {title && (
-      <PanelTitle tag={titleTag}>{title}</PanelTitle>
-    )}
-    {children}
-  </div>
-);
+}) => {
+  let hasSubtextChild = false;
+  let childCount = 0;
+  React.Children.forEach(children, (child) => {
```
Comment (philschanely) [inline packages/sage-react/lib/Panel/PanelHeader.jsx:17]: 😎

**Review by philschanely — APPROVED** [review]

**Review by monicawheeler — APPROVED** [review]
👍🏻

**Review by joedakroub — APPROVED** [review]
Nicely done Ben; no concerns here. 👍🏼

**Review by anechol — APPROVED** [review]
🚀

**Pull request merged**